### DISCUSSION TITLE
Issue migrating from `web3-provider` to `ethereum-provider` using `web3modal1.0` `providerOptions`

### DISCUSSION THREAD

**Discussion by luismmdev** [WalletConnect/walletconnect-monorepo#2642] [Developer Support]
Hi there!

As the title claims, I'm having issues migrating from the deprecated `@walletconnect/web3-provider` to the newer `@walletconnect/ethereum-provider`. In the [docs](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps/#web3-provider) one can read the following:

> **web3-provider**
If you are using `@walletconnect/web3-provider`, we stopped supporting this provider in favor of the improved version published under `@walletconnect/ethereum-provider` and you will be able to check the latest version on NPM [here](https://npmjs.com/package/@walletconnect/ethereum-provider).

However, this is not working as expected. I had the following configuration:

```typescript
const WalletConnect = await import('@walletconnect/web3-provider');
const Fortmatic = await import('fortmatic');
const CoinbaseWallet = await import('@coinbase/wallet-sdk');

const providerOptions = {
    walletconnect: {
      package: WalletConnect.default,
      options: {
        infuraId: '<infuraApiKey>',
      },
    },
    fortmatic: {
      package: Fortmatic.default,
      options: {
        key: '<formaticApiKey>',
      },
    },
    coinbasewallet: {
      package: CoinbaseWallet.default,
      options: {
        infuraId: '<infuraApiKey>',
      },
    }
};

new Web3Modall({
  cacheProvider: true,
  network: 'mainnet',
  providerOptions,
  ...
});
```

Updated the package to the new one and changed:

```typescript
const WalletConnect = await import('@walletconnect/ethereum-provider');
await WalletConnect.EthereumProvider.init({
  projectId: '<projectId>'
  chains: [1],
  showQrModal: true
});

const providerOptions = {
  walletconnect: {
    package: WalletConnect.default,
  },
  ...
};

new Web3Modall({
  cacheProvider: true,
  network: 'mainnet',
  providerOptions,
  ...
});
```

No errors thrown in console. However, I cannot see the `walletconnect` provider in the `web3modal`, but the injected (`metamask`), `formatic` and `coinbasewallet` are working as expected.

What am I doing wrong here? Isn't this suppose to work out of the box just by updating the package as stated in the docs?

Thanks in advance.

Kind regards,
Luis.

**Comment by glitch-txs** [comment]
How are you setting the injected one?

The provider is resolved by EthereumProvider when initializing

Check https://docs.walletconnect.com/2.0/web/providers/ethereum

**Reply by luismmdev** [reply]
Hey @glitch-txs , thanks for answering!

The injected one is automatically detected by web3modal (the `disableInjectedProvider` config option when creating the Web3Modal instance is `false` by default).

Yes, I know the provider is resolved after initialising. However, I want to keep using `web3modal1.0` as the provider selector, because right now there is no similar web3modal available in v2.0. Either I have to migrate completly to the full modal with `wagmi` or users won't be able to connect using metamask's/coinbase/you name it browser extension, as the other web3modal v2.0 alternatives don't support such use case ([as I know you are already aware](https://github.com/orgs/WalletConnect/discussions/2637)).

Is there a way to keep web3modal 1.0 with this very same workflow but just migrate from `web3-provider` to `ethereum-provider`?

**Reply by glitch-txs** [reply]
This is a great missing feature I think, @0xAsimetriq there's no way to add browser wallets to walletconnect/modal right? What Ive seen in the docs asks for links but not sure about providers. I also opened this discussion as suggestion, since this is something I've seen devs to be interested in https://github.com/orgs/WalletConnect/discussions/2637

@luismmdev I'm currently working on a library that uses connectors like wagmi but without viem. If you're interested it's open for collabs 😸

**Reply by luismmdev** [reply]
Sadly it seems so 😞 

Thanks for your suggestions and help. Your work sounds promising! I'm just getting started in the web3 and crypto realm and currently don't have a lot of free time available, but I'll take a look at your library and if I feel I can be useful or contribute in any manner I'll gladly do it!

**Comment by luismmdev** [comment]
Is passing `@walletconnect/ethereum-provider` via `web3modal.providerOptions` deprecated? I think there should be a migrating guide regarding this option, as it seems it is no longer possible but the docs don't say anything about this approach.

**Reply by finessevanes** [reply]
It is deprecated. wagmi has an [Ethers.js Adapters](https://github.com/orgs/WalletConnect/discussions/2637) which allows you to grab the signer/provider.

@glitch-txs we plan on having this in Web3Modal v3! 
https://github.com/orgs/WalletConnect/discussions/2637
